internal/services/controller/controller_test.go

```diff
@@ -242,6 +243,157 @@ func TestController_ApiResourcesErrorProcessing(t *testing.T) {
 	}])
 }
 
+func TestController_ShouldSendByInterval(t *testing.T) {
+	tt := []struct {
+		name          string
+		sendInterval  time.Duration
+		sendDurations []time.Duration
+		checkAfter    time.Duration
+		wantSends     int64
```

Comment:
> Also, I think the test misses a test case where some requests are skipped because it takes longer to send a request than the non-sliding interval.

If I understand you correctly, that would be test case 4 at loc 284:  "should trigger all sends when previous send exceeds two intervals". It checks the sends after 900 ms, which would produce 3 sends if all happened under the 300 ms interval budget, but the first send takes 650 ms, spanning two intervals, so it skips one and produces two.